routes/auth.routes.js

```diff
@@ -0,0 +1,140 @@
+const { Router } = require('express');
+const router = new Router();
+
+const mongoose = require('mongoose');
+const session = require('express-session');
+const MongoStore = require('connect-mongo')(session);
+
+const bcryptjs = require('bcryptjs');
+const saltRounds = 10;
+
+const User = require('../models/User.model');
+ 
+// .get() route ==> to display the signup form to users
+router.get('/signup', (req,res) =>  res.render("./auth/signup"))
+ 
+// .post() route ==> to process form data
+router.post('/signup', (req, res, next) => {
+    const {Username, Password,email} = req.body
+      if (!Username || !Password ){
+      res.render("./auth/signup", {
+          Username,
+          Password,
+          email,
+          errorMessage: "All fields are mandatory. Please provide your username, email and password.",
+      })
+      return;
+      }    
+      
+      const strongPasswordRegex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{4,}/;
+
+  // Validate that incoming password matches regex pattern.
+  if (!strongPasswordRegex.test(Password)) {
+    res.status(500).render("auth/signup", {
+      //email,
+      Username,
+      errorMessage:
+        "Password needs to have at least 5 chars and must contain at least one number, one lowercase and one uppercase letter.",
+    });
+    return;
+  }
+
+
+bcryptjs
+    .genSalt(saltRounds)
+    .then(salt => 
+      bcryptjs.hash(Password,salt))
+    .then((hashedPassword) =>
+      User.create({ Username, email, Password: hashedPassword })
+        .then((newUser) => {
+          console.log(newUser);
```

Comment:
You should have `req.session.user = newUser;` so you can keep track of the user right after the sign up.